packages/odyssey-react-mui/src/Checkbox.tsx

```diff
@@ -90,7 +117,9 @@ const Checkbox = ({
   value,
 }: CheckboxProps) => {
   const { t } = useTranslation();
-  const [isCheckedValue, setIsCheckedValue] = useState(isDefaultChecked);
+  const [isCheckedValue, setIsCheckedValue] = useControlledState(
+    isChecked ?? isDefaultChecked
+  );
```

Comment:
There are 2 ways of doing this:
1. Renaming the prop.
2. Renaming the state.

I've done both.

If we rename the state, I think calling it `isLocalChecked` would make more sense because you're stating this is a local value (as in scoped to your component). Seeing `isCheckedValue` is confusing because `checked` and `value` mean two different things when passed to HTML.

Renaming the prop would be something like `isCheckedProp`, and then comparing that way.